core/jsonparse.mly

```diff
@@ -1,8 +1,15 @@
+(* See Note [Debugging grammar conflicts] in parser.mly *)
+
 %{
 open Utility
 open ProcessTypes
 open WebsocketMessages
 
+module Links_core =
```

Comment:
It would be good to document why this declaration is here in case the workaround needed changes (or Ocaml gets sensible namespaces and we can stop).  Likewise with the other ones.